x-pack/plugins/code/server/indexer/lsp_indexer.ts

```diff
@@ -67,6 +67,47 @@ export class LspIndexer extends AbstractIndexer {
     ];
   }
 
+  protected async *getIndexRequestIterator(): AsyncIterableIterator<LspIndexRequest> {
+    try {
+      const {
+        workspaceRepo,
+        workspaceRevision,
+      } = await this.lspService.workspaceHandler.openWorkspace(this.repoUri, 'head');
+      const workspaceDir = workspaceRepo.workdir();
+      const gitOperator = new GitOperations(this.options.repoPath);
+      const fileIterator = await gitOperator.iterateRepo(this.repoUri, 'head');
+      for await (const file of fileIterator) {
+        const filePath = file.path!;
+        const lang = detectLanguageByFilename(filePath);
+        // filter file by language
+        if (lang && this.lspService.supportLanguage(lang)) {
```

Comment:
it turns out this filtering should happen later. fixed by https://github.com/elastic/kibana/pull/36723